### DISCUSSION TITLE
FPS drop drastically when draw text

### DISCUSSION THREAD

**Discussion by edwinsyarief** [hajimehoshi/ebiten#1531] [Q&A] [answered]
Hi @hajimehoshi ,

I'm new in golang. And I want to learn golang by making games using your library. I'm interested at your library because it is very simple.

But I have an issue when trying to render text on screen, when there is no text, the fps is around 130-140, but when I draw text on screen using ttf, I don't why the fps drop drasctically to 30-40.

Can you help? Here is my current implementation
https://github.com/MozGaming/SuperShooterGame

Thank you.

Best Regards,
Edwin

**Comment by hajimehoshi** [comment, marked as answer]
Hi,

Thank you for using Ebiten. I couldn't see your code in the repository.

Thanks,

**Reply by edwinsyarief** [reply]
Sorry, here is the code
https://github.com/edwinsyarief/supershootergame

Thank you.

**Reply by hajimehoshi** [reply]
https://github.com/edwinsyarief/supershootergame/blob/f7b70bbe8fee4ead739235352ad04d90d2f94fde/core/game.go#L157
You meant that if you remove this line, the FPS changes, right?

**Reply by edwinsyarief** [reply]
> https://github.com/edwinsyarief/supershootergame/blob/f7b70bbe8fee4ead739235352ad04d90d2f94fde/core/game.go#L157
> 
> You meant that if you remove this line, the FPS changes, right?

Yes. Correct.

**Reply by hajimehoshi** [reply]
Sure, let me investigate this later as now I'm working.

**Reply by edwinsyarief** [reply]
Thank you.

**Reply by hajimehoshi** [reply]
I don't see any FPS drops with the text rendering and the app could keep 60 FPS (MacBook Pro 2020). Could you double-check this?

**Reply by edwinsyarief** [reply]
> I don't see any FPS drops with the text rendering and the app could keep 60 FPS (MacBook Pro 2020). Could you double-check this?

Strange, now the fps keep 60, is it related to my code structure?

Thank you

**Reply by hajimehoshi** [reply]
I have no idea but it looks like the issue has been fixed.

Feel free to revisit this thread when you find the same issues again.
